presidio-analyzer/presidio_analyzer/predefined_recognizers/in_vehicle_registration_recognizer.py

```diff
@@ -0,0 +1,1582 @@
+    :param supported_entity: The entity this recognizer can detect
+    :param replacement_pairs: List of tuples with potential replacement values
+        for different strings to be used during pattern matching.
+    This can allow a greater variety in input e.g. by removing dashes or spaces
+    """
+
+    PATTERNS = [
+        Pattern(
+            "India Vehicle Registration (Very Weak)",
+            r"\b[A-Z]{1}(?!0000)[0-9]{4}\b",
+            0.01,
+        ),
+        Pattern(
+            "India Vehicle Registration (Very Weak)",
+            r"\b[A-Z]{2}(?!0000)\d{4}\b",
+            0.01,
+        ),
+        Pattern(
+            "India Vehicle Registration (Very Weak)",
+            r"\b(I)(?!00000)\d{5}\b",
+            0.01,
+        ),
+        Pattern(
+            "India Vehicle Registration (Weak)",
+            r"\b[A-Z]{3}(?!0000)\d{4}\b",
+            0.2,
+        ),
+        Pattern(
+            "India Vehicle Registration (Medium)",
+            r"\b\d{1,3}(CD|CC|UN)[1-9]{1}[0-9]{1,3}\b",
+            0.40,
+        ),
+        Pattern(
+            "India Vehicle Registration",
+            r"\b[A-Z]{2}\d{1,2}[A-Z]{1,2}(?!0000)\d{4}\b",
+            0.85,
+        ),
+        Pattern(
+            "India Vehicle Registration",
+            r"\b[2-9]{1}[1-9]{1}(BH)(?!0000)\d{4}[A-HJ-NP-Z]{2}\b",
+            0.85,
+        ),
+        Pattern(
+            "India Vehicle Registration",
+            r"\b(?!00)\d{2}(A|B|C|D|E|F|H|K|P|R|X)\d{6}[A-Z]{1}\b",
+            0.85,
+        ),
+    ]
+
+    CONTEXT = ["RTO", "vehicle", "plate", "registration"]
+
+    def __init__(
+        self,
+        patterns: Optional[List[Pattern]] = None,
+        context: Optional[List[str]] = None,
+        supported_language: str = "en",
+        supported_entity: str = "IN_VEHICLE_REGISTRATION",
+        replacement_pairs: Optional[List[Tuple[str, str]]] = None,
+    ):
+        self.replacement_pairs = (
+            replacement_pairs
+            if replacement_pairs
+            else [("-", ""), (" ", ""), (":", "")]
+        )
+        patterns = patterns if patterns else self.PATTERNS
+        context = context if context else self.CONTEXT
+        super().__init__(
+            supported_entity=supported_entity,
+            patterns=patterns,
+            context=context,
+            supported_language=supported_language,
+        )
+
+    def validate_result(self, pattern_text: str) -> bool:
+        """Determine absolute value based on calculation."""
+        sanitized_value = Utils.sanitize_value(pattern_text, self.replacement_pairs)
+        # print('Sanitized value:' + sanitized_value)
+        return self.__check_vehicle_registration(sanitized_value)
+
+    def __check_vehicle_registration(self, sanitized_value: str) -> bool:
+        # print('check function called')
+        is_valid_registration = None
+        # logic here
+        an_list = ["01"]
+        ap_list = ["39", "40"]
+        ar_list = [
```

Comment:
Is it possible to create those lists dynamically, and then remove the values that shouldn't be there? It makes the code super long and difficult to go over.

Reply:
Hi @omri374 ,
Yes, the code was becoming unreadable. So , I moved the metadata to analyzer_utils.py. I deliberated on the design before coding it so and went with generic code + hardcoded data approach instead of the other way round to avoid creating too many exceptions making it unmaintanable in the longer run imho. The root cause is lack of regulation in the defined codes resulting in _gaps_ in the series for most of the cases. Plus, there is always a possibility that an entry is **delisted** in future. We will have a tough time incorporating those changes later on. Hence , this approach. Please let me know your opinion on the same.